### PR TITLE
Issue61: returns http status 500 instead of 204 when result body is nil

### PR DESCRIPTION
**Pull request by MachadoLhes**
Preventing issue #61 from happening

### CODE REVIEW

File: src/clojure/restql/http/async_handler.clj [hunk 1]
```diff
@@ -70,10 +70,15 @@
           timeout-ch ([] (log/warn "request handler timed out") (>! error-ch {:status 500 :body "Request timed out"}))
           exception-ch ([err] (>! error-ch (util/error-output err)))
           query-ch ([result]
-                     (log/debug " finishing request handler")
-                     (>! result-ch {:body    (util/format-response-body result)
-                                    :headers (make-headers query-entry result)
-                                    :status  (util/calculate-response-status-code result)})))))
+                       (log/debug " finishing request handler")
+                       (if (nil? result)
+                         (do (>! result-ch {:body    nil
```
Comment (cupello) [inline src/clojure/restql/http/async_handler.clj]: `do` is not necessary.

File: src/clojure/restql/http/async_handler.clj [hunk 2]
```diff
@@ -70,10 +70,15 @@
...
+                                            :headers nil
+                                            :status  500}))
+                         (do (>! result-ch {:body    (util/format-response-body result)
```
Comment (cupello) [inline src/clojure/restql/http/async_handler.clj]: Same here.

File: src/clojure/restql/http/request_util.clj [hunk 3]
```diff
@@ -142,6 +142,7 @@
   (cond
     (nil? a) b
     (nil? b) a
+    (= 204 b) a
```
Comment (cupello) [inline src/clojure/restql/http/request_util.clj]: ???

File: test/clojure/restql/http/request_util_test.clj [hunk 4]
```diff
@@ -119,13 +119,14 @@
...
-          [503 200 404 422 200 408]
+          [503 200 404 422 200 408 200]
           [(higher-value 503 200)
            (higher-value 200 200)
            (higher-value 404 301)
            (higher-value 200 422)
            (higher-value nil 200)
-           (higher-value 408 nil)]))))
+           (higher-value 408 nil)
+           (higher-value 200 204)]))))
```
Comment (cupello) [inline test/clojure/restql/http/request_util_test.clj]: This should be in another PR.

File: src/clojure/restql/http/request_util.clj (hunk 3 above)
Comment (MachadoLhes) [reply src/clojure/restql/http/request_util.clj]: This is the solution to issue #62, probably it ended up here because of a faulty merge of mine

Comment (cupello) [reply src/clojure/restql/http/request_util.clj]: Then it should be removed.
